### PR TITLE
Add new password. 

### PR DESCRIPTION
**Pull request by TsuyoshiUshio**
Obviously dangerous.

### CODE REVIEW

File: VulnerableApplication/Models/Hello.cs [hunk 1]
```diff
@@ -11,5 +11,9 @@ public void Say()
         {
             Console.WriteLine("Hello!");
         }
+    public string Hi() {
+    return "this is against the rules.";
```
Comment (TsuyoshiUshio) [inline VulnerableApplication/Models/Hello.cs:15]: [CODE_SMELL] 2 more lines of code need to be covered by tests to reach the minimum threshold of 65.0% lines coverage.

Comment (TsuyoshiUshio) [inline VulnerableApplication/Models/Hello.cs:15]: [CODE_SMELL] 6 more comment lines need to be written to reach the minimum threshold of 25.0% comment density.

Comment (TsuyoshiUshio) [inline VulnerableApplication/Models/Hello.cs:15]: [CODE_SMELL] Add or update the header of this file.

Comment (TsuyoshiUshio) [inline VulnerableApplication/Models/Hello.cs:15]: [CODE_SMELL] Make 'Hi' a static method.

Comment (TsuyoshiUshio) [inline VulnerableApplication/Models/Hello.cs:15]: [CODE_SMELL] Remove this method and declare a constant for this value.

File: VulnerableApplication/Global.asax.cs [hunk 2]
```diff
@@ -10,12 +10,18 @@ namespace VulnerableApplication
 {
     public class MvcApplication : System.Web.HttpApplication
     {
+        public string Password = "abcdef";
+
```
Comment (TsuyoshiUshio) [inline VulnerableApplication/Global.asax.cs:14]: [CODE_SMELL] 3 more lines of code need to be covered by tests to reach the minimum threshold of 65.0% lines coverage.

Comment (TsuyoshiUshio) [inline VulnerableApplication/Global.asax.cs:14]: [CODE_SMELL] 8 more comment lines need to be written to reach the minimum threshold of 25.0% comment density.

Comment (TsuyoshiUshio) [inline VulnerableApplication/Global.asax.cs:14]: [CODE_SMELL] Add or update the header of this file.

Comment (TsuyoshiUshio) [inline VulnerableApplication/Global.asax.cs:14]: [CODE_SMELL] Make 'Password' private.

Comment (TsuyoshiUshio) [inline VulnerableApplication/Global.asax.cs:14]: [VULNERABILITY] 'password' detected in this expression, review this potentially hardcoded credential.

Comment (TsuyoshiUshio) [inline VulnerableApplication/Global.asax.cs:14]: [VULNERABILITY] Make this field 'private' and encapsulate it in a 'public' property.

File: VulnerableApplication/Models/Hello.cs (hunk 1 above)
Comment (TsuyoshiUshio) [inline VulnerableApplication/Models/Hello.cs:15]: [CODE_SMELL] Make 'Say' a static method.

Comment (TsuyoshiUshio) [inline VulnerableApplication/Models/Hello.cs:15]: [CODE_SMELL] Remove this logging statement.

Comment (TsuyoshiUshio) [inline VulnerableApplication/Models/Hello.cs:15]: [CODE_SMELL] Replace this string literal with a string retrieved through an instance of the 'ResourceManager' class.

Comment (TsuyoshiUshio) [inline VulnerableApplication/Models/Hello.cs:15]: [VULNERABILITY] Remove this logging statement.

File: VulnerableApplication/Controllers/HomeController.cs [hunk 3]
```diff
@@ -8,6 +8,9 @@ namespace VulnerableApplication.Controllers
 {
     public class HomeController : Controller
     {
+        public static string SonarPassword = "foo";
+        public static string GitHubPassword = "bar";
```
Comment (TsuyoshiUshio) [inline VulnerableApplication/Controllers/HomeController.cs:12]: [VULNERABILITY] 'password' detected in this expression, review this potentially hardcoded credential.

Comment (TsuyoshiUshio) [inline VulnerableApplication/Controllers/HomeController.cs:12]: [CODE_SMELL] Change the visibility of 'GitHubPassword' or make it 'const' or 'readonly'.

Comment (TsuyoshiUshio) [inline VulnerableApplication/Controllers/HomeController.cs:12]: [VULNERABILITY] Make this field 'private' and encapsulate it in a 'public' property.

Comment (TsuyoshiUshio) [inline VulnerableApplication/Controllers/HomeController.cs:12]: [CODE_SMELL] Add or update the header of this file.

Comment (TsuyoshiUshio) [inline VulnerableApplication/Controllers/HomeController.cs:12]: [CODE_SMELL] 4 more lines of code need to be covered by tests to reach the minimum threshold of 65.0% lines coverage.

Comment (TsuyoshiUshio) [inline VulnerableApplication/Controllers/HomeController.cs:12]: [CODE_SMELL] 10 more comment lines need to be written to reach the minimum threshold of 25.0% comment density.

Comment (TsuyoshiUshio) [inline VulnerableApplication/Controllers/HomeController.cs:12]: [VULNERABILITY] 'password' detected in this expression, review this potentially hardcoded credential.

Comment (TsuyoshiUshio) [inline VulnerableApplication/Controllers/HomeController.cs:12]: [CODE_SMELL] Change the visibility of 'SonarPassword' or make it 'const' or 'readonly'.

Comment (TsuyoshiUshio) [inline VulnerableApplication/Controllers/HomeController.cs:12]: [VULNERABILITY] Make this field 'private' and encapsulate it in a 'public' property.

File: VulnerableApplication/Models/Hello.cs (hunk 1 above)
Comment (TsuyoshiUshio) [inline VulnerableApplication/Models/Hello.cs:15]: [CODE_SMELL] 2 more lines of code need to be covered by tests to reach the minimum threshold of 65.0% lines coverage.

Comment (TsuyoshiUshio) [inline VulnerableApplication/Models/Hello.cs:15]: [CODE_SMELL] 6 more comment lines need to be written to reach the minimum threshold of 25.0% comment density.

Comment (TsuyoshiUshio) [inline VulnerableApplication/Models/Hello.cs:15]: [CODE_SMELL] Add or update the header of this file.

Comment (TsuyoshiUshio) [inline VulnerableApplication/Models/Hello.cs:15]: [CODE_SMELL] Make 'Hi' a static method.

Comment (TsuyoshiUshio) [inline VulnerableApplication/Models/Hello.cs:15]: [CODE_SMELL] Remove this method and declare a constant for this value.

File: VulnerableApplication/Controllers/HomeController.cs (hunk 3 above)
Comment (TsuyoshiUshio) [inline VulnerableApplication/Controllers/HomeController.cs:12]: [CODE_SMELL] Remove this method and declare a constant for this value.

File: VulnerableApplication/Global.asax.cs (hunk 2 above)
Comment (TsuyoshiUshio) [inline VulnerableApplication/Global.asax.cs:14]: [CODE_SMELL] 3 more lines of code need to be covered by tests to reach the minimum threshold of 65.0% lines coverage.

Comment (TsuyoshiUshio) [inline VulnerableApplication/Global.asax.cs:14]: [CODE_SMELL] 8 more comment lines need to be written to reach the minimum threshold of 25.0% comment density.

Comment (TsuyoshiUshio) [inline VulnerableApplication/Global.asax.cs:14]: [CODE_SMELL] Add or update the header of this file.

Comment (TsuyoshiUshio) [inline VulnerableApplication/Global.asax.cs:14]: [CODE_SMELL] Make 'Password' private.

Comment (TsuyoshiUshio) [inline VulnerableApplication/Global.asax.cs:14]: [VULNERABILITY] 'password' detected in this expression, review this potentially hardcoded credential.

Comment (TsuyoshiUshio) [inline VulnerableApplication/Global.asax.cs:14]: [VULNERABILITY] Make this field 'private' and encapsulate it in a 'public' property.

File: VulnerableApplication/Models/Hello.cs (hunk 1 above)
Comment (TsuyoshiUshio) [inline VulnerableApplication/Models/Hello.cs:15]: [CODE_SMELL] Make 'Say' a static method.

Comment (TsuyoshiUshio) [inline VulnerableApplication/Models/Hello.cs:15]: [CODE_SMELL] Remove this logging statement.

Comment (TsuyoshiUshio) [inline VulnerableApplication/Models/Hello.cs:15]: [CODE_SMELL] Replace this string literal with a string retrieved through an instance of the 'ResourceManager' class.

Comment (TsuyoshiUshio) [inline VulnerableApplication/Models/Hello.cs:15]: [VULNERABILITY] Remove this logging statement.

File: VulnerableApplication/Controllers/HomeController.cs (hunk 3 above)
Comment (TsuyoshiUshio) [inline VulnerableApplication/Controllers/HomeController.cs:12]: [VULNERABILITY] 'password' detected in this expression, review this potentially hardcoded credential.

Comment (TsuyoshiUshio) [inline VulnerableApplication/Controllers/HomeController.cs:12]: [CODE_SMELL] Change the visibility of 'GitHubPassword' or make it 'const' or 'readonly'.

Comment (TsuyoshiUshio) [inline VulnerableApplication/Controllers/HomeController.cs:12]: [VULNERABILITY] Make this field 'private' and encapsulate it in a 'public' property.

Comment (TsuyoshiUshio) [inline VulnerableApplication/Controllers/HomeController.cs:12]: [CODE_SMELL] Add or update the header of this file.

Comment (TsuyoshiUshio) [inline VulnerableApplication/Controllers/HomeController.cs:12]: [CODE_SMELL] 4 more lines of code need to be covered by tests to reach the minimum threshold of 65.0% lines coverage.

Comment (TsuyoshiUshio) [inline VulnerableApplication/Controllers/HomeController.cs:12]: [CODE_SMELL] 10 more comment lines need to be written to reach the minimum threshold of 25.0% comment density.

Comment (TsuyoshiUshio) [inline VulnerableApplication/Controllers/HomeController.cs:12]: [VULNERABILITY] 'password' detected in this expression, review this potentially hardcoded credential.

Comment (TsuyoshiUshio) [inline VulnerableApplication/Controllers/HomeController.cs:12]: [CODE_SMELL] Change the visibility of 'SonarPassword' or make it 'const' or 'readonly'.

Comment (TsuyoshiUshio) [inline VulnerableApplication/Controllers/HomeController.cs:12]: [VULNERABILITY] Make this field 'private' and encapsulate it in a 'public' property.

File: VulnerableApplication/Models/Hello.cs (hunk 1 above)
Comment (TsuyoshiUshio) [inline VulnerableApplication/Models/Hello.cs:15]: [CODE_SMELL] 2 more lines of code need to be covered by tests to reach the minimum threshold of 65.0% lines coverage.

Comment (TsuyoshiUshio) [inline VulnerableApplication/Models/Hello.cs:15]: [CODE_SMELL] 6 more comment lines need to be written to reach the minimum threshold of 25.0% comment density.

Comment (TsuyoshiUshio) [inline VulnerableApplication/Models/Hello.cs:15]: [CODE_SMELL] Add or update the header of this file.

Comment (TsuyoshiUshio) [inline VulnerableApplication/Models/Hello.cs:15]: [CODE_SMELL] Make 'Hi' a static method.

Comment (TsuyoshiUshio) [inline VulnerableApplication/Models/Hello.cs:15]: [CODE_SMELL] Remove this method and declare a constant for this value.

File: VulnerableApplication/Controllers/HomeController.cs (hunk 3 above)
Comment (TsuyoshiUshio) [inline VulnerableApplication/Controllers/HomeController.cs:12]: [CODE_SMELL] Remove this method and declare a constant for this value.

File: VulnerableApplication/Global.asax.cs [hunk 4]
```diff
@@ -10,12 +10,19 @@ namespace VulnerableApplication
...
```
Comment (TsuyoshiUshio) [inline VulnerableApplication/Global.asax.cs:14]: [CODE_SMELL] 3 more lines of code need to be covered by tests to reach the minimum threshold of 65.0% lines coverage.

Comment (TsuyoshiUshio) [inline VulnerableApplication/Global.asax.cs:14]: [CODE_SMELL] 7 more comment lines need to be written to reach the minimum threshold of 25.0% comment density.

Comment (TsuyoshiUshio) [inline VulnerableApplication/Global.asax.cs:14]: [CODE_SMELL] Add or update the header of this file.

Comment (TsuyoshiUshio) [inline VulnerableApplication/Global.asax.cs:14]: [CODE_SMELL] Make 'Password' private.

Comment (TsuyoshiUshio) [inline VulnerableApplication/Global.asax.cs:14]: [VULNERABILITY] 'password' detected in this expression, review this potentially hardcoded credential.

Comment (TsuyoshiUshio) [inline VulnerableApplication/Global.asax.cs:14]: [VULNERABILITY] Make this field 'private' and encapsulate it in a 'public' property.

File: VulnerableApplication/Models/Hello.cs (hunk 1 above)
Comment (TsuyoshiUshio) [inline VulnerableApplication/Models/Hello.cs:15]: [CODE_SMELL] Make 'Say' a static method.

Comment (TsuyoshiUshio) [inline VulnerableApplication/Models/Hello.cs:15]: [CODE_SMELL] Remove this logging statement.

Comment (TsuyoshiUshio) [inline VulnerableApplication/Models/Hello.cs:15]: [CODE_SMELL] Replace this string literal with a string retrieved through an instance of the 'ResourceManager' class.

Comment (TsuyoshiUshio) [inline VulnerableApplication/Models/Hello.cs:15]: [VULNERABILITY] Remove this logging statement.

Comment (TsuyoshiUshio) [reply VulnerableApplication/Models/Hello.cs:15]: /workitem

Comment (TsuyoshiUshio) [reply VulnerableApplication/Models/Hello.cs:15]: /workitem

Comment (TsuyoshiUshio) [reply VulnerableApplication/Models/Hello.cs:15]: /workitem

Comment (TsuyoshiUshio) [reply VulnerableApplication/Models/Hello.cs:15]: /workitem

File: VulnerableApplication/Global.asax.cs (hunk 2 above)
Comment (TsuyoshiUshio) [reply VulnerableApplication/Global.asax.cs:14]: /workitem

Comment (TsuyoshiUshio) [reply VulnerableApplication/Global.asax.cs:14]: /workitem

Comment (TsuyoshiUshio) [reply VulnerableApplication/Global.asax.cs:14]: /workitem

Comment (TsuyoshiUshio) [reply VulnerableApplication/Global.asax.cs:14]: /workitem

Comment (TsuyoshiUshio) [reply VulnerableApplication/Global.asax.cs:14]: /workitem

Comment (TsuyoshiUshio) [reply VulnerableApplication/Global.asax.cs:14]: /workitem

Comment (TsuyoshiUshio) [reply VulnerableApplication/Global.asax.cs:14]: /workitem

Comment (TsuyoshiUshio) [reply VulnerableApplication/Global.asax.cs:14]: /workitem

Comment (TsuyoshiUshio) [reply VulnerableApplication/Global.asax.cs:14]: /workitem

Comment (TsuyoshiUshio) [reply VulnerableApplication/Global.asax.cs:14]: /workitem

Comment (TsuyoshiUshio) [reply VulnerableApplication/Global.asax.cs:14]: /workitem

Comment (TsuyoshiUshio) [reply VulnerableApplication/Global.asax.cs:14]: /workitem

Comment (TsuyoshiUshio) [reply VulnerableApplication/Global.asax.cs:14]: /workitem

Comment (TsuyoshiUshio) [reply VulnerableApplication/Global.asax.cs:14]: /workitem

Comment (TsuyoshiUshio) [reply VulnerableApplication/Global.asax.cs:14]: /workitem

Comment (TsuyoshiUshio) [reply VulnerableApplication/Global.asax.cs:14]: /workitem

Comment (TsuyoshiUshio) [reply VulnerableApplication/Global.asax.cs:14]: /workitem

Comment (TsuyoshiUshio) [reply VulnerableApplication/Global.asax.cs:14]: /workitem
